queries/unix_filespace.kusto

```diff
@@ -1,21 +1,34 @@
 // more or equal than 90% Diskspace used (Unix)
-let mountpoints = dynamic(["/", "/var", "/boot", "/tmp", "/home", "/opt", "/backup"]);
+let mountpoints = dynamic(["/", "/var", "/boot", "/tmp", "/home", "/opt", "/backup", "/boot/efi"]);
 let warning = int(80);
 let critical = int(90);
 Perf
-    | where ObjectName contains "Logical Disk"
-    | where CounterName == "% Used Space"
-    | where InstanceName in (mountpoints)
-    | where CounterValue >= (warning)
-    | where TimeGenerated > ago(5m)
-    | extend state = case(CounterValue >= (critical), "Critical", CounterValue <= (critical), "Warning", "")
-    | extend threshold = case(CounterValue >= (critical), critical, CounterValue < (critical), warning, warning)
-    | extend monitor_package = "AZ_UX_SC_ManagedOSUnix"
-    | extend monitor_name = "AZ_UNIX_FILESPACE"
-    | extend monitor_description = strcat("Disk threshold reached. More or equal than ", (threshold), "% Diskspace used. Check the Diskspace of the Mountpoint, please.")
-    | extend script_name = "n/a"
-    | extend script_version = "n/a"
-    | extend value = round(CounterValue, 2)
-    | extend additional_information = strcat("more or equal than ", (threshold), "% Diskspace used (Unix)")
-    | extend affected_entity = InstanceName
-    | project TimeGenerated, _ResourceId, state, affected_object = Computer, monitor_package, monitor_name, monitor_description, script_name, script_version, threshold, value, affected_entity, additional_information
+| where ObjectName contains "Logical Disk"
+| where CounterName == "% Used Space"
+| where InstanceName in (mountpoints)
+| where CounterValue >= (warning)
+| where TimeGenerated > ago(5m)
+| extend affected_object = toupper(tostring(split(_ResourceId, "/") [-1]))
+| summarize arg_max(TimeGenerated, *) by affected_object, InstanceName
+| extend _ResourceId = toupper(_ResourceId)
+| project _ResourceId, affected_object, CounterValue, InstanceName
+| join kind= leftouter (MonitoringResources_CL | where TimeGenerated > ago(13h)
+| where type_s == "microsoft.compute/virtualmachines" 
+| extend MonitoringResources = toupper(tostring(split(id_s, "/") [-1])) 
+| extend id_s = toupper(id_s)
+| project TimeGenerated,MonitoringResources, id_s, name_s, tags_managedby_s
+| summarize arg_max(TimeGenerated,*) by MonitoringResources)
```

Comment:
This is the actual table we are joining right? 
I would replace that by a "let" table to increase readability.

Reply:
Yes, this is the actual table we are joining.  I'm going to change the join part. I'm replacing that by a let definition.